test/new-e2e/pkg/components/remotehost_waitforagent.go

```diff
@@ -0,0 +1,193 @@
+		}
+
+		agents := map[string]func() (string, bool){
+			"process-agent":  params.processAgentStatusEndpoint,
+			"trace-agent":    params.traceAgentStatusEndpoint,
+			"security-agent": params.securityAgentStatusEndpoint,
+		}
+
+		for name, ep := range agents {
+			endpoint, ok := ep()
+			if !ok {
+				continue
+			}
+
+			h.context.T().Logf("Checking status of the %s...", name)
+			cmd := curlCommand(h.OSFamily, endpoint, params.authToken)
+			_, err := h.Execute(cmd)
+
+			if !assert.NoErrorf(t, err, "%s did not become ready", name) {
+				h.context.T().Logf("Failed to check %s status: %s", name, err)
+			}
+		}
+	}, params.waitFor, params.tick, "Waiting for agents to be ready")
+}
+
+// WithAuthToken sets the auth token.
+func WithAuthToken(authToken string) AgentReadyOption {
+	return func(p *agentReadyParams) {
+		p.authToken = authToken
+	}
+}
+
+// WithAuthTokenPath sets the path to the auth token file.
+// The file is read from the remote host.
+// This is not used if the auth token is provided directly with WithAuthToken.
+func WithAuthTokenPath(path string) AgentReadyOption {
+	return func(p *agentReadyParams) {
+		p.authTokenPath = path
+	}
+}
+
+// WithProcessAgentOnPort enables waiting for the Process Agent, using the given port for the API.
+func WithProcessAgentOnPort(port int) AgentReadyOption {
+	return func(p *agentReadyParams) {
+		p.processAgentPort = port
+	}
+}
+
+// WithProcessAgent enables waiting for the Process Agent, using the default API port.
+func WithProcessAgent() AgentReadyOption {
+	return WithProcessAgentOnPort(6162)
+}
+
+// WithTraceAgentOnPort enables waiting for the Trace Agent, using the given port for the API.
+func WithTraceAgentOnPort(port int) AgentReadyOption {
+	return func(p *agentReadyParams) {
+		p.traceAgentPort = port
+	}
+}
+
+// WithTraceAgent enables waiting for the Trace Agent, using the default API port.
+func WithTraceAgent() AgentReadyOption {
+	return WithTraceAgentOnPort(5012)
+}
+
+// WithSecurityAgentOnPort enables waiting for the Security Agent, using the given port for the API.
+func WithSecurityAgentOnPort(port int) AgentReadyOption {
+	return func(p *agentReadyParams) {
+		p.securityAgentPort = port
+	}
+}
+
+// WithSecurityAgent enables waiting for the Security Agent, using the default API port.
+func WithSecurityAgent() AgentReadyOption {
+	return WithSecurityAgentOnPort(5010)
+}
+
+// WithWaitFor sets the duration to wait for the agents to be ready.
+func WithWaitFor(d time.Duration) AgentReadyOption {
+	return func(p *agentReadyParams) {
+		p.waitFor = d
+	}
+}
+
+// WithTick sets the duration between checks for the agents to be ready.
+func WithTick(d time.Duration) AgentReadyOption {
+	return func(p *agentReadyParams) {
+		p.tick = d
+	}
+}
+
+func (p *agentReadyParams) processAgentStatusEndpoint() (string, bool) {
+	if p.processAgentPort == 0 {
+		return "", false
+	}
+	return fmt.Sprintf("http://localhost:%d/agent/status", p.processAgentPort), true
```

Comment:
**❓ question:** what's the difference between `agent/status` and `/config/all` you mentioned in another PR ?‏

Reply:
Basically in the other PR I needed to fetch the config from every process so I used that endpoint directly, but this PR is more general so I used the status endpoints of each agent